R/plot.R

```diff
@@ -111,58 +111,51 @@ plot_ethnicity <- function(x) {
+    labs(y = "Daily number of patient attendances", x = "Arrival date", colour = "Visit type") +
     theme(legend.position = "bottom") +
-    ggtitle("Admission profile by type of admission")
+    ggtitle("Admission profile by type of admission")+
+    scale_color_manual(values = c("#379481",
+                                  "#4CA1CB",
+                                  "#3AB8EE"))
 }
 
-
-
-#' Plot Transition Matrix
+#' Plot Visit detail Profile
 #'
-#' @param tm a transition matrix from \code{\link{transition_matrix}}
-#' @param possible_states a character vector of the possible states to be used
-#'   in the markov chain fitting process
+#' @param x
 #'
-#' @importFrom tibble as_tibble add_column
-#' @importFrom tidyr pivot_longer
-#' @importFrom dplyr mutate across
-#' @importFrom ggplot2 ggplot aes geom_tile geom_text theme scale_fill_viridis_c
-#'   ggtitle
-#' @importFrom magrittr %>%
+#' @importFrom dplyr mutate_at vars group_by tally
+#' @importFrom ggplot2 ggplot aes geom_path theme labs
 #'
 #' @return
 #' @export
 #'
 #' @examples
-plot_tm <- function(tm, possible_states) {
-
-  tm_fit <- as_tibble(tm$estimate@transitionMatrix) %>%
-    add_column(
-      source = rownames(tm$estimate@transitionMatrix), .before = TRUE) %>%
-    pivot_longer(-contains("source"),
-                 names_to = "destination",
-                 values_to = "value") %>%
-    mutate(across(c("source", "destination"), factor, levels = possible_states))
-
-  tm_fit %>%
-    ggplot(aes(y = source, x = destination)) +
-    geom_tile(aes(fill = value)) +
-    geom_text(aes(label = round(value, 2)), colour = "white") +
+plot_visit_detail_profile <- function(x) {
+  x %>%
+    mutate_at(vars(visit_detail_start_datetime), ~ as.Date(.)) %>%
+    mutate_at(vars(visit_detail_concept_id), ~ as.character(.)) %>%
+    # Shouldn't really hardcode these dates. Will edit as we set tolerance for data quality report.
+    filter(visit_detail_start_datetime > "2017-01-01" & visit_detail_start_datetime <= Sys.Date()) %>% 
+    group_by(visit_detail_start_datetime, visit_detail_concept_id) %>%
+    tally() %>%
+    ggplot(aes(x = visit_detail_start_datetime, y = n,
+               group = visit_detail_concept_id, colour = visit_detail_concept_id)) +
+    geom_path() +
     theme_d() +
-    theme(
-      plot.title.position = "plot",
-      panel.grid.major = element_blank(),
-      legend.position = "none",
-      axis.text.x = element_text(angle = 35, hjust = 1)) +
-    scale_fill_viridis_c() +
-    ggtitle("Transition Matrix for Internal Hospital Movements")
+    labs(y = "Daily number of patients in visit details", x = "Date", colour = "Visit type") +
+    theme(legend.position = "bottom") +
+    ggtitle("Daily observation profile by type of observation")+
+    scale_color_manual(values = c("#379481",
+                                  "#4CA1CB",
+                                  "#3AB8EE"))
 }
 
 #' DECOVID Plot Theme
```

Comment:
Is there a plot theme?
Where is it?

Reply:
It is used in plots, (age plot, sex plot,...). Shall I remove ' DECOVID'.? This is from Ed.